components/PinSetup.tsx

```diff
@@ -22,7 +22,7 @@ export function PinSetup({ mode = 'setup' }: PinSetupProps) {
   const [errorMessage, setErrorMessage] = useState('');
 
   const handleCurrentPinComplete = async (pin: string) => {
-    const isValid = await AuthService.verifyPin(pin);
+    const isValid = await AuthService.verifyPinLegacy(pin);
```

Comment:
I'm not really following how you're validating the first and second PIN values, but you shouldn't be calling the auth service at all until this point.

This suggests to me you're preemtively saving the initial PIN before it's even been confirmed. But what happens if the app terminates before the confirmation? You'll have saved the initial PIN.

Also, it's an overkill to execute the PIN validation routine during the PIN setup anyway.